### ISSUE TITLE
RJ_SCENE_SAVE prepare message

### ISSUE THREAD

**Issue by danomatika** [rjdj/rjlib#22]
I have an abstraction that uses a table to store sequences that I'm saving and loading via [u_dispatch] objects. I ran into an issue where I need to know when the state is being saved but *before* the actual "save" message is sent out in order to dump the contents of the table into [u_dispatch]. Otherwise, it's too late to dump the table by the time the "save" message is received.

I'd like to propose adding a "prepare" message to RJ_SCENE_SAVE and sending out just before sending the "save" message. I'm doing the following and the is working well: [prepare, save<.

**Comment by footils** [comment]
Hi,

hm, I'm not (yet) convinced that it would be necessary to do any
changes to u_dispatch/u_loader/u_sssad. Maybe it's indeed better solved
outside as you seem to have done?

You already can send "prepare" to RJ_SCENE_SAVE if you like. Just listen
on RJ_SCENE_SAVE, route out "prepare" to do any preparations like
dumping the table. Then send "save" after "prepare".

Or maybe I don't understand how you propose to add the "prepare" message ...

All the best
-- 
Frank

On Tue, Nov 15, 2016 at 07:28:34PM -0800, Dan Wilcox wrote:
> I have an abstraction that uses a table to store sequences that I'm
> saving and loading via [u_dispatch] objects. I ran into an issue where
> I need to know when the state is being saved but *before* the actual
> "save" message is sent out in order to dump the contents of the table
> into [u_dispatch]. Otherwise, it's too late to dump the table by the
> time the "save" message is received.
> 
> I'd like to propose adding a "prepare" message to RJ_SCENE_SAVE and
> sending out just before sending the "save" message. I'm doing the
> following and the is working well: [prepare, save<.
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/rjdj/rjlib/issues/22

-- 
 Frank Barknecht                                     _ ______footils.org__

**Comment by danomatika** [comment]
Right. No changes are directly necessary. I'm proposing that a "prepare" message be added to the documentation as this is a general problem that is applicable beyond my set of rjlib compatible abstractions.

**Comment by footils** [comment]
Hi,

I see. That's a good idea. I'll write something up to add to
u_loader-help.pd - or do you already have something?

Ciao
-- 
Frank

On Wed, Nov 16, 2016 at 03:33:12AM -0800, Dan Wilcox wrote:
> Right. No changes are directly necessary. I'm proposing that a "prepare" message be added to the documentation as this is a general problem that is applicable beyond my set of rjlib compatible abstractions.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/rjdj/rjlib/issues/22#issuecomment-260923988

-- 
 Frank Barknecht                                     _ ______footils.org__
